model/connect/MySQLStatement.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * Provides a record-view for mysqli statements
+ *
+ * By default streams unbuffered data, but seek(), rewind(), or numRecords() will force the statement to
+ * buffer itself and sacrifice any potential performance benefit.
+ */
+class MySQLStatement extends SS_Query {
+
+	/**
+	 * The related mysqli statement object if generated using a prepared query
+	 *
+	 * @var mysqli_stmt
+	 */
+	protected $statement;
+
+	/**
+	 * Metadata result for this statement
+	 *
+	 * @var mysqli_result
+	 */
+	protected $metadata;
+
+	/**
+	 * Is the statement bound to the current resultset?
+	 *
+	 * @var bool
+	 */
+	protected $bound = false;
+
+	/**
+	 * List of column names
+	 *
+	 * @var array
+	 */
+	protected $columns = array();
+
+	/**
+	 * List of bound variables in the current row
+	 *
+	 * @var array
+	 */
+	protected $boundValues = array();
+
+	/**
+	 * Binds this statement to the variables
+	 */
+	protected function bind() {
+		$variables = array();
+
+		// Bind each field
+		while($field = $this->metadata->fetch_field()) {
+			$this->columns[] = $field->name;
+			$variables[] = &$this->boundValues[$field->name];
```

Comment:
This part looks weird because at this point, boundValues is empty. By using the reference operator, it doesn't need to be isset, since it simply allows the statement to write to this bound value on `$statement->fetch()`. It's weird code because mysqli relies on results being bound by reference to a statement.